src/muc_light/mod_muc_light.erl

```diff
@@ -138,66 +164,60 @@ delete_room(RoomUS) ->
+    ejabberd_hooks:add(hooks(HostType)),
+    %% Handler
+    SubdomainPattern = subdomain_pattern(HostType),
+    PacketHandler = mongoose_packet_handler:new(?MODULE),
+    mongoose_domain_api:register_subdomain(HostType, SubdomainPattern, PacketHandler),
+    ok.
 
-    mod_muc_light_db_backend:start(Host, MUCHost),
-    mongoose_subhosts:register(Host, MUCHost),
-    ejabberd_router:register_route(MUCHost, mongoose_packet_handler:new(?MODULE)),
-
-    %% Prepare config schema
-    ConfigSchema = mod_muc_light_room_config:schema_from_definition(
-                     gen_mod:get_opt(config_schema, Opts, default_schema_definition())),
-    gen_mod:set_module_opt(Host, ?MODULE, config_schema, ConfigSchema),
-
-    %% Prepare default config
-    DefaultConfig = mod_muc_light_room_config:default_from_schema(ConfigSchema),
-    gen_mod:set_module_opt(Host, ?MODULE, default_config, DefaultConfig),
-
+-spec stop(HostType :: host_type()) -> ok.
+stop(HostType) ->
+    SubdomainPattern = subdomain_pattern(HostType),
+    mongoose_domain_api:unregister_subdomain(HostType, SubdomainPattern),
+    mod_muc_light_db_backend:stop(HostType),
+    ejabberd_hooks:delete(hooks(HostType)),
     ok.
 
--spec stop(Host :: jid:server()) -> ok.
-stop(Host) ->
-    MUCHost = gen_mod:get_module_opt_subhost(Host, ?MODULE, default_host()),
-    ejabberd_router:unregister_route(MUCHost),
-    mongoose_subhosts:unregister(MUCHost),
+%% Init helpers
+subdomain_pattern(HostType) ->
+    gen_mod:get_module_opt(HostType, ?MODULE, host, default_host()).
 
-    mod_muc_light_db_backend:stop(Host, MUCHost),
+server_host_to_muc_host(HostType, ServerHost) ->
+    mongoose_subdomain_utils:get_fqdn(subdomain_pattern(HostType), ServerHost).
 
-    ejabberd_hooks:delete(hooks(Host, MUCHost)),
+host_type_to_codec(HostType) ->
+    case gen_mod:get_module_opt(HostType, ?MODULE, legacy_mode, ?DEFAULT_LEGACY_MODE) of
+        true ->
+            legacy;
+        false ->
+            modern
+    end.
 
-    %% Hook for room in roster
-    ejabberd_hooks:delete(roster_get, Host, ?MODULE, add_rooms_to_roster, 50),
-    %% Hooks for legacy mode
-    ejabberd_hooks:delete(privacy_iq_get, Host, ?MODULE, process_iq_get, 1),
-    ejabberd_hooks:delete(privacy_iq_set, Host, ?MODULE, process_iq_set, 1),
+tracked_db_funs() ->
+    [create_room, destroy_room, room_exists, get_user_rooms,
+     remove_user, remove_domain,
+     get_config, set_config, get_blocking, set_blocking,
+     get_aff_users, modify_aff_users].
 
-    ok.
+set_dynamic_opts(HostType, Opts) ->
+    %% Prepare config schema
+    Def = gen_mod:get_opt(config_schema, Opts, default_schema_definition()),
+    ConfigSchema = mod_muc_light_room_config:schema_from_definition(Def),
+    %% XXX That is a bad style
+    gen_mod:set_module_opt(HostType, ?MODULE, computed_config_schema, ConfigSchema),
+    %% Prepare default config
+    DefaultConfig = mod_muc_light_room_config:default_from_schema(ConfigSchema),
```

Comment:
What is this, and why is it a bad style?

Reply:
modules should not set their own options.